.ci/magician/github/integration_test.go

```diff
@@ -0,0 +1,310 @@
+//go:build integration
```

Comment:
I'm a little hesitant about this file. I see it's using build tags so it won't build by default, and we're not running it in CI, but it's not a common pattern for us.

Could you add a README file explaining how this is intended to be used?

Reply:
added a readme, I am okay removing it if you want.

Reply:
I think let's keep it for now.